azure_li_services/units/user.py

```diff
@@ -86,7 +86,12 @@ def create_or_modify_user(user):
     user_exists = system_users.user_exists(user['username'])
     if 'group' in user:
         if not system_users.group_exists(user['group']):
```

Comment:
This is not sufficient. 

`group_exists()`

Searches for the name. However, if the name does not exists we will try to add the specified id, which also triggers and error. For example:

```
# groupadd -g 1000 foo
groupadd: GID '1000' already exists
```

So either group_exists() needs to search for both "name" and "id" or we need two functions, group_name_exists() and group_id_exists()

Reply:
I disagree, you can't create a group if no name is given. group_exists() is called only if a group name is specified in the config file. This is ensured by ```if 'group' in user:``` The optional group_id information is only taken into account if also a group name is specified.

>So either group_exists() needs to search for both "name" and "id" or we need two functions, >group_name_exists() and group_id_exists()

no on Linux you never can have a group without a name. The search is based on the name and
will always find if that group exists no matter which id is assigned to it

The way this is described in the schema is not optimal as it would also allow to specify a group_id with no group name which will result in pointless information because without a group name we don't do anything in the code.

The necessary schema change to formulate that would be an incompatible change to the existing schema which I did not implement because I said we are feature complete. Thus a further schema change will only be done if this is ok from the Microsoft side. The change would then look like this:

```yaml
group:
    name: some-group-name[mandatory]
    id: some-group-id[optional]
```

former yaml will then no longer be valid. @jeffaco please comment

Thanks

Reply:
As I'm pretty sure we want it clean I will add a commit to show what I mean